pkg/proxy/k8s.go

```diff
@@ -72,15 +74,25 @@ func (kc *KubernetesClient) InitClient(address, token string, gateway *model.Gat
 		address = ReplaceURLHostAndPort(originUrl, "127.0.0.1", proxyAddr.Port)
 	}
 
-	kubeConfig := kc.GetKubeConfig(address, token)
+	kubeConf := &rest.Config{
+		Host:        address,
+		BearerToken: token,
+	}
+	kubeConf.Insecure = true
+
+	if !srvconn.IsValidK8sUserToken(kubeConf) {
+		return srvconn.ErrValidToken
+	}
+
+	kubeConfigYAML := kc.GetKubeConfig(address, token)
 
 	tmpFile, err := os.CreateTemp("", "kubeconfig-*.yaml")
 	if err != nil {
 		return fmt.Errorf("error creating temp file: %w", err)
 	}
 	defer tmpFile.Close()
 
-	if _, err := tmpFile.Write([]byte(kubeConfig)); err != nil {
+	if _, err := tmpFile.Write([]byte(kubeConfigYAML)); err != nil {
 		return fmt.Errorf("error writing to temp file: %w", err)
 	}
 	kc.configName = tmpFile.Name()
```

Comment:
The above code provides different syntax changes between two versions of the same function, which is not relevant for the comparison. Please provide another question related to this topic for an appropriate analysis and correction.